core/src/test/java/org/elasticsearch/search/internal/InternalSearchHitTests.java

```diff
@@ -19,24 +19,143 @@
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
 import org.elasticsearch.index.Index;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.search.SearchHitField;
 import org.elasticsearch.search.SearchShardTarget;
+import org.elasticsearch.search.fetch.subphase.highlight.HighlightField;
+import org.elasticsearch.search.fetch.subphase.highlight.HighlightFieldTests;
+import org.elasticsearch.search.internal.InternalSearchHit.InternalNestedIdentity;
 import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.RandomObjects;
 
+import java.io.IOException;
 import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.Set;
 
+import static org.elasticsearch.common.xcontent.XContentHelper.toXContent;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertToXContentEquivalent;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.notNullValue;
 import static org.hamcrest.Matchers.nullValue;
 
 public class InternalSearchHitTests extends ESTestCase {
 
+    private static Set<String> META_FIELDS = Sets.newHashSet("_uid", "_all", "_parent", "_routing", "_size", "_timestamp", "_ttl");
+
+    public static InternalSearchHit createTestItem(boolean withOptionalInnerHits) {
+        int internalId = randomInt();
+        String uid = randomAsciiOfLength(10);
+        Text type = new Text(randomAsciiOfLengthBetween(5, 10));
+        InternalNestedIdentity nestedIdentity = null;
+        if (randomBoolean()) {
+            nestedIdentity = InternalNestedIdentityTests.createTestItem(randomIntBetween(0, 2));
+        }
+        Map<String, SearchHitField> fields = new HashMap<>();
+        if (randomBoolean()) {
+            int size = randomIntBetween(0, 10);
+            for (int i = 0; i < size; i++) {
+                Tuple<List<Object>, List<Object>> values = RandomObjects.randomStoredFieldValues(random(),
+                        XContentType.JSON);
+                if (randomBoolean()) {
+                    String metaField = randomFrom(META_FIELDS);
+                    fields.put(metaField, new InternalSearchHitField(metaField, values.v1()));
+                } else {
+                    String fieldName = randomAsciiOfLength(5);
+                    fields.put(fieldName, new InternalSearchHitField(fieldName, values.v1()));
+                }
+            }
+        }
+        InternalSearchHit hit = new InternalSearchHit(internalId, uid, type, nestedIdentity, fields);
+        if (frequently()) {
+            hit.score(randomFloat());
+        }
+        if (frequently()) {
+            hit.sourceRef(RandomObjects.randomSource(random()));
+        }
+        if (randomBoolean()) {
+            hit.version(randomLong());
```

Comment:
Thinking about it again, version is settable, so we might want to test what happens if it is not set (in which case the default should be used). Maybe this doesn't happen in out code just yet, but version() setter might not be called after all.